### PR TITLE
[Py3] fix various issues in tests related to BytesWarning

### PR DESCRIPTION
**Pull request by MartinBasti**

### CODE REVIEW

File: ipapython/ipaldap.py [hunk 1]
```diff
@@ -379,8 +379,10 @@ def _set_raw(self, name, value):
                 name, value.__class__.__name__, value))
         for (i, item) in enumerate(value):
             if not isinstance(item, bytes):
-                raise TypeError("%s[%d] value must be str, got %s object %r" % (
-                    name, i, item.__class__.__name__, item))
+                raise TypeError(
+                    "%s[%d] value must be bytes, got %s object %r" % (
+                        name, i, item.__class__.__name__, item)
```
Comment (MartinBasti) [inline ipapython/ipaldap.py]: nitpick, leave it consistent with other code in that module, can be refactored at once later

Comment (stlaz) [inline ipapython/ipaldap.py]: Ok.

**Comment by MartinBasti** [comment]
The last commit has wrong ticket

**Comment by MartinBasti** [comment]
Fixed upstream
master:
https://fedorahosted.org/freeipa/changeset/d38540acd614bcaa489023401fc8db7c02cd3892
https://fedorahosted.org/freeipa/changeset/6bb5af7bea21d44b4e5ee20cfaa2f76b12ea0929
https://fedorahosted.org/freeipa/changeset/a5ccdc16cbcec433ef061dfe65515e32c3021ea2

**Pull request closed without merging**